themes/classic/templates/_partials/microdata/product-jsonld.tpl

```diff
@@ -0,0 +1,107 @@
+      "reviewCount": "{$nbComments|escape:'html':'UTF-8'}"
+    },{/if}
+    {if isset($product.weight) && ($product.weight != 0)}
+      "weight": {
+        "@context": "https://schema.org",
+        "@type": "QuantitativeValue",
+        "value": "{$product.weight}",
+        "unitCode": "{$product.weight_unit}"
+      },{/if}
+      {if $product.show_price}
+        {if $product.id_product_attribute == 0}
+          "offers": {
+            "@type": "Offer",
+            "priceCurrency": "{$currency.iso_code}",
+            "name": "{$product.name|strip_tags:false}",
+            "price": "{$product.price_amount}",
+            "url": "{$product.url}",
+            "priceValidUntil": "{($smarty.now + (int) (60*60*24*15))|date_format:"%Y-%m-%d"}",
+            {if $product.images|count > 0}
+              "image": {strip}[
+                {foreach from=$product.images item=p_img name="p_img_list"}
+                  "{$p_img.large.url}"{if not $smarty.foreach.p_img_list.last},{/if}
+                {/foreach}
+              ]{/strip},
+            {/if}
+            "sku": "{if $product.reference}{$product.reference}{else}{$product.id}{/if}",
+            "mpn": "{if $product.mpn}{$product.mpn}{elseif $product.reference}{$product.reference}{else}{$product.id}{/if}",
+            {if $product.condition == 'new'}"itemCondition": "https://schema.org/NewCondition",{/if}
+            {if $product.show_condition > 0}
+              {if $product.condition == 'used'}"itemCondition": "https://schema.org/UsedCondition",{/if}
+              {if $product.condition == 'refurbished'}"itemCondition": "https://schema.org/RefurbishedCondition",{/if}
+            {/if}
+            "availability": "{if $product.quantity > 0 || $product.allow_oosp > 0}https://schema.org/InStock{else}https://schema.org/OutOfStock{/if}",
+            "seller": {
+              "@type": "Organization",
+              "name": "{$shop.name}"
+            }
+          }
+        {else}
+          "offers": [
+              {foreach key=id_product_combination item=combination from=$combinations}
+                {
+                  "@type": "Offer",
+                  "priceCurrency": "{$currency.iso_code}",
+                  "name": "{$product.name|strip_tags:false} - {foreach item=attr from=$combination.attributes_values}{$attr|strip_tags:false}
+                    {if !$attr@last} 
+                    {/if}
+                  {/foreach}",
+                  "price": "{Product::getPriceStatic($product->id, true, $id_product_combination)|round:'2'}",
+                  "url": "{$product.url}",
+                  "priceValidUntil": "{($smarty.now + (int) (60*60*24*15))|date_format:"%Y-%m-%d"}",
+                  "image": "{if $combination.id_image > 0}{$link->getImageLink($product->link_rewrite, $combination.id_image, 'home_default')|escape:'html':'UTF-8'}{else}{$product.cover.bySize.home_default.url}{/if}",
+                  "mpn": "{$combination.reference}",
```

Comment:
You can now use proper MPN here and also add GTIN here 🚀

Reply:
Yes, to avoid unnecessary Google warning I added mpn like for product. (if exists go for it, if not take reference). Google want sku and at least one identifier (so mpn, because it's considered as a sort of internal id and gtin and isbn, nice to have if defined).

Reply:
@Hlavtox if you can check if you defined isbn,ean13 and upc for combination in your PR, that would be super great ;)

Reply:
@fdonnet I did only ean and mpn, sorry man. Made a new PR: https://github.com/PrestaShop/PrestaShop/pull/23094, after it's merged you can use it.